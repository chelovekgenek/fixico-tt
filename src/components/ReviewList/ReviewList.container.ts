import { connect } from "react-redux";

import { ReviewList } from "./ReviewList";

import { TAppState } from "store/entities";
import {
  getItems,
  ReviewsDeleteActions,
  getPagination,
  ReviewsPaginationActions
} from "store/entities/reviews";

export interface IStateProps {
  reviews: ReturnType<typeof getItems>;
  pagination: ReturnType<typeof getPagination>;
}
export interface IDispatchProps {
  deleteById: typeof ReviewsDeleteActions.request;
  setPage: typeof ReviewsPaginationActions.setPage;
  setLimit: typeof ReviewsPaginationActions.setLimit;
}

export const ReviewListContainer = connect<
  IStateProps,
  IDispatchProps,
  {},
  TAppState
>(
  state => ({
    reviews: getItems(state),
    pagination: getPagination(state)
  }),
  {
    deleteById: ReviewsDeleteActions.request,
    setPage: ReviewsPaginationActions.setPage,
    setLimit: ReviewsPaginationActions.setLimit
  }
)(ReviewList);
