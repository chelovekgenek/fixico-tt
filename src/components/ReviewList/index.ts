export { ReviewListContainer as ReviewList } from "./ReviewList.container";
