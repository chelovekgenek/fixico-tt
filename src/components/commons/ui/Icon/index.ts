export { Icon } from "@material-ui/core";
