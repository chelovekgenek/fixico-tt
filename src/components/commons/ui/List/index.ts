import * as MUI from "@material-ui/core";

import { Text } from "./Item";
import { Avatar } from "./Avatar";

export const List = Object.assign(MUI.List, {
  Item: Object.assign(MUI.ListItem, {
    Text,
    Avatar,
    SecondaryAction: MUI.ListItemSecondaryAction
  })
});
