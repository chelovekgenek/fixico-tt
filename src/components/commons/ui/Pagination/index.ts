export { TablePagination as Pagination } from "@material-ui/core";
