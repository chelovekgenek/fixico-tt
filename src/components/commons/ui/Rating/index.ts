export { Rating } from "@material-ui/lab";
