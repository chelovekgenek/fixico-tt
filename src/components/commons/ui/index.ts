export * from "./Button";
export * from "./IconButton";
export * from "./Icon";
export * from "./List";
export * from "./Pagination";
export * from "./Rating";
