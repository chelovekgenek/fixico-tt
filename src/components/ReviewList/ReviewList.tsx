import React, { useCallback, ComponentProps } from "react";

import {
  List,
  IconButton,
  Icon,
  Rating,
  Pagination
} from "components/commons/ui";

import { IStateProps, IDispatchProps } from "./ReviewList.container";

interface IProps extends IStateProps, IDispatchProps {}

export const ReviewList: React.FC<IProps> = ({
  reviews,
  pagination,
  deleteById,
  setPage,
  setLimit
}) => {
  const handleChangePage = useCallback<
    React.ComponentProps<typeof Pagination>["onChangePage"]
  >((e, page) => setPage(page), [setPage]);
  const handleChangeRowsPerPage = useCallback<
    Required<React.ComponentProps<typeof Pagination>>["onChangeRowsPerPage"]
  >(e => setLimit(Number(e.target.value)), [setLimit]);
  return (
    <React.Fragment>
      <List>
        {reviews.map(item => (
          <List.Item key={item.id}>
            <List.Item.Avatar>
              <Rating value={item.rating} readOnly />
            </List.Item.Avatar>
            <List.Item.Text>{item.body}</List.Item.Text>
            <List.Item.SecondaryAction>
              <IconButton
                edge="end"
                aria-label="delete"
                onClick={() => deleteById(item.id)}
              >
                <Icon>delete</Icon>
              </IconButton>
            </List.Item.SecondaryAction>
          </List.Item>
        ))}
      </List>
      {Boolean(pagination.total) && (
        <Pagination
          rowsPerPageOptions={[5, 10, 20]}
          component="div"
          count={pagination.total}
          rowsPerPage={pagination.limit}
          page={pagination.page}
          backIconButtonProps={{
            "aria-label": "previous page"
          }}
          nextIconButtonProps={{
            "aria-label": "next page"
          }}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      )}
    </React.Fragment>
  );
};
