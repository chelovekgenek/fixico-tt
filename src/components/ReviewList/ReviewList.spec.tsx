import React from "react";
import { shallow } from "enzyme";

import { Button, IconButton } from "components/commons/ui";
import reviews from "mocks/reviews.json";
import pagination from "mocks/pagination.json";

import { ReviewList } from "./ReviewList";

describe("ReviewForm", () => {
  const render = (
    props: Partial<React.ComponentProps<typeof ReviewList>> = {}
  ) =>
    shallow(
      <ReviewList
        reviews={reviews}
        pagination={pagination}
        deleteById={jest.fn() as any}
        setPage={jest.fn() as any}
        setLimit={jest.fn() as any}
        {...props}
      />
    );
  it("should render", () => {
    expect(render()).toMatchSnapshot();
  });
  it("shouldn't render anything if data is not provided", () => {
    expect(
      render({ reviews: [], pagination: { ...pagination, total: 0 } })
    ).toMatchSnapshot();
  });
  it("should call delete function", () => {
    const props = { deleteById: jest.fn() as any };
    const comp = render(props);
    comp
      .find(IconButton)
      .at(0)
      .simulate("click");
    expect(props.deleteById).toBeCalledWith(reviews[0].id);
  });
});
