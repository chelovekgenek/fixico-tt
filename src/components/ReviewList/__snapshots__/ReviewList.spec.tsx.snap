// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm should render 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      key="1"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={3}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        hey something's going on!
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="6"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={5}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        test the 6th comment
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="7"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={2}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        something's going on
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="8"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={2}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        tjyjfghjfghjf
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="9"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={3}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        ygfhkfjkff
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="10"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={3}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        TESTING LOADING...
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="11"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={5}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        something's going on
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="12"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={3}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        fasgfasdgasgasdgasdg
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      key="15"
    >
      <Styled(WithStyles(ForwardRef(ListItemAvatar)))>
        <WithStyles(ForwardRef(Rating))
          readOnly={true}
          value={4}
        />
      </Styled(WithStyles(ForwardRef(ListItemAvatar)))>
      <Styled(WithStyles(ForwardRef(ListItemText)))>
        fgsdgfdfgdgfdsdfdgs
      </Styled(WithStyles(ForwardRef(ListItemText)))>
      <WithStyles(ForwardRef(ListItemSecondaryAction))>
        <WithStyles(ForwardRef(IconButton))
          aria-label="delete"
          edge="end"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(Icon))>
            delete
          </WithStyles(ForwardRef(Icon))>
        </WithStyles(ForwardRef(IconButton))>
      </WithStyles(ForwardRef(ListItemSecondaryAction))>
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(TablePagination))
    backIconButtonProps={
      Object {
        "aria-label": "previous page",
      }
    }
    component="div"
    count={5}
    nextIconButtonProps={
      Object {
        "aria-label": "next page",
      }
    }
    onChangePage={[Function]}
    onChangeRowsPerPage={[Function]}
    page={1}
    rowsPerPage={5}
    rowsPerPageOptions={
      Array [
        5,
        10,
        20,
      ]
    }
  />
</Fragment>
`;

exports[`ReviewForm shouldn't render anything if data is not provided 1`] = `
<Fragment>
  <WithStyles(ForwardRef(List)) />
</Fragment>
`;
